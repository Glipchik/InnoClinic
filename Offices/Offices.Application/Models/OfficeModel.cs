namespace Offices.Application.Models
{
    public class OfficeModel
    {
        public required Guid Id { get; set; }
        public required string Address { get; set; }
        public required string RegistryPhoneNumber { get; set; }
        public required bool IsActive { get; set; }
        public required IEnumerable<DoctorModel> Doctors { get; set; }
        public required IEnumerable<ReceptionistModel> Receptionists { get; set; }
    }
}
