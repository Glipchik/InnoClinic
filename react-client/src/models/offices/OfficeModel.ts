export default interface OfficeModel {
  id: string
  address: string
  registryPhoneNumber: string
  isActive: boolean
}