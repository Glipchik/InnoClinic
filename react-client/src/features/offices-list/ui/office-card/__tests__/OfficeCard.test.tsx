import { render, waitFor } from "@testing-library/react";
import OfficeModel from "@shared/models/offices/officeModel";
import { MemoryRouter } from "react-router-dom";
import OfficeCard from "..";
import { mockOfficeModel, mockOnDelete } from "./__mocks__";
import userEvent from "@testing-library/user-event";

const renderOfficeCard = (office: OfficeModel, onDelete: () => void) =>
  render(
    <MemoryRouter>
      <OfficeCard item={office} onDelete={onDelete} />
    </MemoryRouter>
  );

describe("OfficeCard", () => {
  it("renders office card with correct data", async () => {
    const { getByTestId, getByText } = renderOfficeCard(
      mockOfficeModel,
      mockOnDelete
    );

    await waitFor(() => {
      expect(getByTestId("office-card")).toBeInTheDocument();
      expect(getByText("Address: 123 Main St")).toBeInTheDocument();
      expect(getByText("Phone: +1234567890")).toBeInTheDocument();
      expect(getByText("Is active: Yes")).toBeInTheDocument();
      expect(getByTestId("edit-button")).toBeInTheDocument();
      expect(getByTestId("deactivate-button")).toBeInTheDocument();
    });
  });

  it("calls onDelete when deactivate button is clicked", async () => {
    const { getByTestId } = renderOfficeCard(mockOfficeModel, mockOnDelete);

    await waitFor(async () => {
      const deactivateButton = getByTestId("deactivate-button");
      await userEvent.click(deactivateButton);
      expect(mockOnDelete).toHaveBeenCalledTimes(1);
    });
  });

  it("disables deactivate button when office is not active", async () => {
    const inactiveOffice: OfficeModel = {
      ...mockOfficeModel,
      isActive: false,
    };

    const { getByTestId } = renderOfficeCard(inactiveOffice, mockOnDelete);

    await waitFor(async () => {
      const deactivateButton = getByTestId("deactivate-button");
      expect(deactivateButton).toBeDisabled();
    });
  });

  it("renders edit button with correct link", async () => {
    const { getByTestId } = renderOfficeCard(mockOfficeModel, mockOnDelete);

    await waitFor(async () => {
      const editButton = getByTestId("edit-button");
      expect(editButton.closest("a")).toHaveAttribute("href", "/offices/edit/1");
    });
  });
});
