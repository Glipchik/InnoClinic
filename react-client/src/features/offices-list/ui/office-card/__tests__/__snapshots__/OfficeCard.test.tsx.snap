// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfficeCard matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="flex flex-col p-6 bg-white rounded-lg shadow-lg hover:shadow-2xl transition-shadow"
    data-testid="office-card"
  >
    <div
      class="flex flex-col space-y-3"
    >
      <div>
        <h3
          class="text-xl font-semibold"
        >
          Address: 123 Main St
        </h3>
      </div>
      <div
        class="flex flex-col space-y-1"
      >
        <p
          class="text-lg"
        >
          Phone: +1234567890
        </p>
        <p
          class="text-lg"
        >
          Is active: Yes
        </p>
      </div>
      <div
        class="flex justify-between mt-4"
      >
        <a
          data-discover="true"
          href="/offices/edit/1"
        >
          <button
            class="px-4 py-2 bg-blue-500 text-white rounded disabled:bg-blue-900 disabled:text-gray-400 disabled:hover:cursor-not-allowed hover:bg-blue-600 focus:outline-none cursor-pointer undefined"
            data-testid="edit-button"
          >
            Edit
          </button>
        </a>
        <button
          class="px-4 py-2 bg-blue-500 text-white rounded disabled:bg-blue-900 disabled:text-gray-400 disabled:hover:cursor-not-allowed hover:bg-blue-600 focus:outline-none cursor-pointer bg-red-600 hover:bg-red-700 disabled:bg-red-800 disabled:text-gray-100"
          data-testid="deactivate-button"
        >
          Deactivate
        </button>
      </div>
    </div>
  </li>
</DocumentFragment>
`;
